File: Desktop/optirh/src/main/java/tn/nexus/Entities/transport/Vehicule.java
package tn.nexus.Entities.transport;

public class Vehicule {

    private int id;
    private String disponibilite;
    private String type;
    private int nbrplace;
    private int trajetId;
    private int nbrReservation;


    public Vehicule() {
    }

    public Vehicule(int id, String disponibilite, String type, int nbrplace, int trajetId, int nbrReservation) {
        this.id = id;
        this.disponibilite = disponibilite;
        this.type = type;
        this.nbrplace = nbrplace;
        this.trajetId = trajetId;
        this.nbrReservation = 0;

    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getDisponibilite() {
        return disponibilite;
    }

    public void setDisponibilite(String disponibilite) {
        this.disponibilite = disponibilite;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public int getNbrplace() {
        return nbrplace;
    }

    public void setNbrplace(int nbrplace) {
        this.nbrplace = nbrplace;
    }
    public int getTrajetId() {
        return trajetId;
    }

    public void setTrajetId(int trajetId) {
        this.trajetId = trajetId;
    }

    public int getNbrReservation() {
        return nbrReservation;
    }

    public void setNbrReservation(int nbrReservation) {
        this.nbrReservation = nbrReservation;
    }

    @Override
    public String toString() {
        return "Reservation_Evenement{" +
                "id=" + id +
                ", disponibilite=" + disponibilite +
                ", type=" + type +
                ", nbrplace=" + nbrplace +
                ", trajetId=" + trajetId +
                '}';
    }
}

File: Web/optiRH/src/Entity/Transport/Vehicule.php
<?php

namespace App\Entity\Transport;

use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;

#[ORM\Entity(repositoryClass: "App\Repository\Transport\VehiculeRepository")]
class Vehicule
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private $id;

    
    #[ORM\Column(type: 'string', length: 255)]
    #[Assert\NotBlank(message: 'La disponibilité est obligatoire')]
    private $disponibilite;

    #[ORM\Column(type: 'string', length: 255)]
    #[Assert\NotBlank(message: 'Le type de véhicule est obligatoire')]
    private $type;

    #[ORM\Column(type: 'integer')]
    #[Assert\NotBlank(message: 'Le nombre de places est obligatoire')]
    #[Assert\Positive(message: 'Le nombre de places doit être positif')]
    private $nbrplace;

    #[ORM\ManyToOne(targetEntity: Trajet::class, inversedBy: 'vehicules')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Trajet $trajet;

    #[ORM\OneToMany(mappedBy: 'vehicule', targetEntity: ReservationTrajet::class, cascade: ['persist', 'remove'], orphanRemoval: true)]
    private Collection $reservations;

    #[ORM\Column(type: 'integer')]
    private $nbrReservation = 0;

    public function __construct()
    {
        $this->reservations = new ArrayCollection();
    }

    // Ajoutez ici TOUS les getters et setters pour chaque propriété
    public function getId(): ?int
    {
        return $this->id;
    }

    public function getDisponibilite(): ?string
    {
        return $this->disponibilite;
    }

    public function setDisponibilite(string $disponibilite): self
    {
        $this->disponibilite = $disponibilite;
        return $this;
    }

// Ajoutez ces méthodes à votre classe Vehicule

public function getType(): ?string
{
    return $this->type;
}

public function setType(string $type): self
{
    $this->type = $type;
    return $this;
}

public function getNbrplace(): ?int
{
    return $this->nbrplace;
}

public function setNbrplace(int $nbrplace): self
{
    $this->nbrplace = $nbrplace;
    return $this;
}

public function getTrajet(): ?Trajet
{
    return $this->trajet;
}

public function setTrajet(?Trajet $trajet): self
{
    $this->trajet = $trajet;
    return $this;
}

public function getNbrReservation(): ?int
{
    return $this->nbrReservation;
}

public function setNbrReservation(int $nbrReservation): self
{
    $this->nbrReservation = $nbrReservation;
    return $this;
}

/**
 * @return Collection|ReservationTrajet[]
 */
public function getReservations(): Collection
{
    return $this->reservations;
}

public function addReservation(ReservationTrajet $reservation): self
{
    if (!$this->reservations->contains($reservation)) {
        $this->reservations[] = $reservation;
        $reservation->setVehicule($this);
    }

    return $this;
}

public function removeReservation(ReservationTrajet $reservation): self
{
    if ($this->reservations->removeElement($reservation)) {
        // set the owning side to null (unless already changed)
        if ($reservation->getVehicule() === $this) {
            $reservation->setVehicule(null);
        }
    }

    return $this;
}}